help_test.go

```diff
@@ -101,6 +101,39 @@ func Test_Help_Custom_Flags(t *testing.T) {
 	}
 }
 
+func Test_Help_Required_Flags_Does_Not_Error(t *testing.T) {
+	oldFlag := HelpFlag
+	defer func() {
+		HelpFlag = oldFlag
+	}()
+
+	app := App{
+		Flags: []Flag{
+			&StringFlag{Name: "foo", Required: true},
+		},
+	}
+
+	output := new(bytes.Buffer)
```

Comment:
If we don't need to test the output here for any reason, we can just use `ioutil.Discard`